src/ophys_etl/modules/segmentation/merge/roi_merging.py

```diff
@@ -0,0 +1,378 @@
+
+        local_t0 = time.time()
+        self_corr_lookup = create_self_corr_lookup(
+                               merger_candidates,
+                               sub_video_lookup,
+                               timeseries_lookup,
+                               filter_fraction,
+                               n_processors)
+
+        new_merger_metrics = get_merger_metric_from_pairs(
+                                   merger_candidates,
+                                   sub_video_lookup,
+                                   timeseries_lookup,
+                                   self_corr_lookup,
+                                   filter_fraction,
+                                   n_processors)
+
+        logger.info('updated self_corr and calculated metrics in '
+                    f'{time.time()-local_t0:.2f} seconds')
+
+        for pair in new_merger_metrics:
+            metric = new_merger_metrics[pair]
+            if metric >= -1.0*corr_acceptance:
+                merger_to_metric[pair] = metric
+
+        potential_mergers = []
+        merger_metrics = []
+        for pair in merger_to_metric:
+            potential_mergers.append(pair)
+            merger_metrics.append(merger_to_metric[pair])
+
+        # order mergers by metric in descending order
+        potential_mergers = np.array(potential_mergers)
+        merger_metrics = np.array(merger_metrics)
+        sorted_indices = np.argsort(-1*merger_metrics)
+        potential_mergers = potential_mergers[sorted_indices]
+
+        recently_merged = set()
+        wait_for_it = set()
+
+        area_lookup = {}
+        for roi_id in roi_lookup:
+            area_lookup[roi_id] = roi_lookup[roi_id].area
+
+        for merger in potential_mergers:
+            roi_id_0 = merger[0]
+            roi_id_1 = merger[1]
+            go_ahead = True
+
+            # if one of the ROIs no longer exists because
+            # it was absorbed into another, skip
+            if roi_id_0 not in valid_roi_id:
+                go_ahead = False
+            if roi_id_1 not in valid_roi_id:
+                go_ahead = False
+
+            # if one of the ROIs has grown by 10% or more since
+            # the start of this iteration, skip
+            if go_ahead:
+                if roi_id_0 in recently_merged:
+                    if roi_lookup[roi_id_0].area > 1.1*area_lookup[roi_id_0]:
+                        go_ahead = False
+                if roi_id_1 in recently_merged:
+                    if roi_lookup[roi_id_1].area > 1.1*area_lookup[roi_id_1]:
+                        go_ahead = False
+
+            # if one of the ROIs was in a potential pair that was skipped
+            # for one of the reasons above, skip (in case that merger
+            # really was the best merger for it; these mergers will
+            # be reconsidered in the next iteration)
+            if go_ahead:
+                if roi_id_0 in wait_for_it:
```

Comment:
can be written more succinctly as 
```
go_ahead = roi_id_0 in wait_for_it or roi_id_1 in wait_for_it
```

Reply:
For me, the code is more readable as it is. If you feel strongly about this, I can change it.